### DISCUSSION TITLE
What is the correct way to name  relations? Are there guidelines or conventions I should follow?

### DISCUSSION THREAD

**Discussion by lumenwrites** [prisma/prisma#10395] [Q&A] [answered]
I have a `User` model which can post `Comment`s and vote on them. `Comment`s can also be parented to each other (for nested comments like on HN or Reddit`).

```prisma
model User {
  [...]
  // Comments the user has posted
  comments Comment[] @relation("UserToComment")
  // Comments the user has voted on
  upvotedComments Comment[] @relation("_UpvoterToComment")
  downvotedComments Comment[] @relation("_DownvoterToComment")  
}

model Comment {
  [...]
  // Comment's author
  author User @relation("UserToComment", fields: [authorId], references: [id])
  authorId String

  // Users who voted on the comment
  upvoters User[] @relation("_UpvoterToComment")
  downvoters User[] @relation("_DownvoterToComment")
  
  // One-to-many self-relation for nested comments
  parentId String?
  parent Comment? @relation("Parent", fields: [parentId], references: [id])
  children Comment[] @relation("Parent")
}
```

I have a few questions on how to correctly name these relations.

How should relation that describes the `User` being an author of several `Comments` be named:
- "UserToComment"?
- "Author"?
- "CommentAuthor"?
- "UserComments"?

I've read somewhere that many-to-many relation names are supposed to start with an underscore (but I'm not sure why, and whether this applies to one-to-many relations as well). How should the relation that describes the multiples `User`s being upvoters of multiple `Comments` be named:
- "_UpvoterToComment"?
- "UpvoterToComment"?
- "UserUpvoted"?
- "Upvoters"?

Finally, I have a `Comment` relating to itself for the nested comments. Should this relation be named:
- "Parent"?
- "Children"?
- "ParentChildren"?
- "CommentToComment"?
- Any of the above but with the underscore in front?

Can you guys help me to understand the correct conventions here, give me some advice on how should name my relations?

**Comment by austincrim** [comment, marked as answer]
Hey @lumenwrites 👋!

We don't have any hard rules about naming but my colleague offered a few guidelines:
1. I would choose a name that semantically describes the relation
1. I also think it’s helpful to include both model names
1. The underscore is used by Prisma to name the relation table in implicit many-to-many relations but it’s not required for relation names

At the end of the day, it's up to personal preference and established team norms.

Let me know if you have any more questions!

**Reply by lumenwrites** [reply]
Thank you, this is helpful!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
